File: frontend/Product-Management-System/src/app/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Orders } from './orders';
import { Product } from './product';

@Injectable({
  providedIn: 'root'
})
export class OrderService {
  baseLink = "http://localhost:8080/api/order/v1/"
  constructor(private httpClient:HttpClient) { }
  getAllOrders():Observable<Orders[]>{
    return this.httpClient.get<Orders[]>(`${this.baseLink+"getAllOrders"}`);
  }
  getAllProductsOfOrder(orderId:number):Observable<Product[]>{
    return this.httpClient.get<Product[]>(`${this.baseLink+"getAllProductsOfOrder/"}${orderId}`);
  }
  getOrderById(orderId:number):Observable<Orders>{
    return this.httpClient.get<Orders>(`${this.baseLink+"getOrderById/"}${orderId}`);
  }
  addOrder(order:Orders):Observable<Orders>{
    return this.httpClient.post<Orders>(`${this.baseLink+"addOrder"}`,order);
  }//assignOrderWithProduct/{orderId}/product/{productSerialNumber}
  assignOrderWithProduct(orderId:number,productSerialNumber:string):Observable<Orders>{
    return this.httpClient.put<Orders>(`${this.baseLink}assignOrderWithProduct/${orderId}/product/${productSerialNumber}`,null);
  }
  deleteOrderById(id:number):Observable<Boolean>{
    return this.httpClient.delete<Boolean>(`${this.baseLink}deleteOrderById/${id}`);
  }
  
}

File: backend/products/src/main/java/com/spring/products/service/OrderService.java
package com.spring.products.service;

import com.spring.products.entity.Orders;
import com.spring.products.entity.Product;

import java.util.List;
import java.util.Set;

public interface OrderService {
    public Orders addOrder(Orders order);
    public Orders updateOrder(long orderId,Orders order);
    public List<Orders>getAllOrders();
    public Orders getOrderById(long orderId);
    public Boolean deleteOrderById(long orderId,String reasonReturn);
    public Orders assignOrderWithProduct(long orderId,long productId);
    public List<Product> getAllProductsOfOrder(long orderId);
}
